java-checks/src/main/java/org/sonar/java/checks/IgnoredTestsCheck.java

```diff
@@ -20,17 +20,34 @@
 package org.sonar.java.checks;
 
 import com.google.common.collect.ImmutableList;
+import java.util.List;
 import org.sonar.check.Rule;
+import org.sonar.java.checks.helpers.ConstantUtils;
+import org.sonar.java.matcher.MethodMatcher;
+import org.sonar.java.matcher.MethodMatcherCollection;
 import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.semantic.SymbolMetadata;
+import org.sonar.plugins.java.api.tree.Arguments;
+import org.sonar.plugins.java.api.tree.BlockTree;
+import org.sonar.plugins.java.api.tree.ExpressionStatementTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
 import org.sonar.plugins.java.api.tree.MethodTree;
 import org.sonar.plugins.java.api.tree.Tree;
 
-import java.util.List;
-
 @Rule(key = "S1607")
 public class IgnoredTestsCheck extends IssuableSubscriptionVisitor {
 
+  private static final String ORG_JUNIT_ASSUME = "org.junit.Assume";
+  private static final String BOOLEAN_TYPE = "boolean";
+
+  private static final MethodMatcherCollection ASSUME_METHODS = MethodMatcherCollection.create(
+    MethodMatcher.create().typeDefinition(ORG_JUNIT_ASSUME).name("assumeTrue").parameters(BOOLEAN_TYPE),
+    MethodMatcher.create().typeDefinition(ORG_JUNIT_ASSUME).name("assumeTrue").parameters("java.lang.String", BOOLEAN_TYPE),
+    MethodMatcher.create().typeDefinition(ORG_JUNIT_ASSUME).name("assumeFalse").parameters(BOOLEAN_TYPE),
+    MethodMatcher.create().typeDefinition(ORG_JUNIT_ASSUME).name("assumeFalse").parameters("java.lang.String", BOOLEAN_TYPE)
```

Comment:
1) there's no test of assumeTrue/assumeFalse with a description
2) consistency: If `@Ignore` with a description is compliant, assumeTrue/assumeFalse with a description should be compliant, so I would remove the method matchers with 2 arguments

Reply:
Agreed with second point, i remove those 2 methods.